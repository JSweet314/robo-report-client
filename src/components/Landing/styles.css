.landing {
  height: 100%;
  min-height: 50vh;
  padding: 1em;
  margin: auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.landing__h2 {
  width: fit-content;
  font-size: 4rem;
  margin: 1rem auto;
}

.landing__h4 {
  width: 60%;
  margin: .2em auto;
}

.landing__text-block {
  width: 80%;
  display: flex;
  justify-content: space-evenly;
}

.landing__p {
  width: 70%;
  margin: 1rem;
  line-height: 1.2;
  font-size: 2rem;
}

.landing__stats-group {
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  width: 90%;
  border: 5px solid lightgray;
  margin: 1rem 0;
}

.stats-group {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  font-size: 4rem;
  padding: 2rem;
}

.stats-group p {
  text-align: center;
  border-bottom: 1px solid black;
  padding: 0 0 1rem 0;
}

.stats-group__total {
  font-size: 6rem;
  margin: 1rem 0;
}


