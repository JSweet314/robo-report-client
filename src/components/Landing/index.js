import React from 'react';
import PropTypes from 'prop-types';
import Instructions from '../Instructions';
import './styles.css';

const Landing = ({ user }) => {
  const welcomeText = user.id ? `Welcome Back ${user.firstName}!` :
    'Welcome to Robo Report!';

  return (
    <div className="landing">
      <h2 className='landing__h2'>{welcomeText}</h2>
      {
        user.id ? <Instructions /> : (
          <div>
            <p className='landing__p'>
              Robo calls have become the primary incoming phone calls to our 
              mobile devices. It&apos;s incredibly frustrating. However, the 
              annoyance of fielding and blocking those calls has not superseded 
              the annoyance of finding and filling out the FCC complaint form.
            </p>
            <p className='landing__p'>
              RoboReport is here to help.<br /> Create an account with our app, 
              and it will auto-populate the FCC complaint form, greatly reducing
              the primary barrier to getting this complaint data to the FCC, and
              hopefully, spurring the FCC to take action to stop these
              incessant, unwanted calls.
            </p>
          </div>
        )
      }
    </div>
  );
};

Landing.propTypes = {
  user: PropTypes.object.isRequired
};

export default Landing;
