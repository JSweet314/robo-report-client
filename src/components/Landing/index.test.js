import React from 'react';
import { shallow } from 'enzyme';
import Landing from './';

describe('Landing', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Landing isLoggedIn={false}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot', () => {
    wrapper = shallow(<Landing isLoggedIn={true}/>);

    expect(wrapper).toMatchSnapshot();
  });
});