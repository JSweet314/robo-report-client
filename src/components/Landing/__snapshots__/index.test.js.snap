// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match the snapshot 1`] = `
<div
  className="landing"
>
  <h2
    className="landing__h2"
  >
    Welcome to Robo Report!
  </h2>
  <div>
    <p
      className="landing__p"
    >
      Robo calls have become the primary incoming phone calls to our mobile devices. It's incredibly frustrating. However, the annoyance of fielding and blocking those calls has not superseded the annoyance of finding and filling out the FCC complaint form.
    </p>
    <p
      className="landing__p"
    >
      RoboReport is here to help.
      <br />
       Create an account with our app, and it will auto-populate the FCC complaint form, greatly reducing the primary barrier to getting this complaint data to the FCC, and hopefully, spurring the FCC to take action to stop these incessant, unwanted calls.
    </p>
  </div>
  <BarChart
    fccData={Array []}
  />
</div>
`;

exports[`Landing should match the snapshot 2`] = `
<div
  className="landing"
>
  <h2
    className="landing__h2"
  >
    Welcome Back Walter!
  </h2>
  <Instructions />
  <BarChart
    fccData={Array []}
  />
</div>
`;
